java-manta-cli/src/main/java/com/joyent/manta/cli/MantaCLI.java

```diff
@@ -17,364 +19,367 @@
+        @CommandLine.Option(names = {"-h", "--help"}, help = true)
+        private boolean isHelpRequested;
 
-        b.append("Creating new connection object").append(BR);
-        try (MantaClient client = new MantaClient(config)) {
-            b.append(INDENT).append(client).append(BR);
+        @CommandLine.Option(names = {"--log-level"},
+                            description = "TRACE, DEBUG, INFO, WARN(default), ERROR")
+        private CommandLogLevel logLevel;
 
-            String homeDirPath = config.getMantaHomeDirectory();
-            b.append("Attempting HEAD request to: ").append(homeDirPath).append(BR);
 
-            MantaObjectResponse response = client.head(homeDirPath);
-            b.append(INDENT).append(response).append(BR);
-            b.append("Request was successful");
+        public abstract void run();
+
+        /**
+         * Builds a new {@link ConfigContext} instance based on defaults and
+         * environment variables.
+         *
+         * @return chained configuration context object
+         */
+        protected ConfigContext buildConfig() {
+            return new ChainedConfigContext(new DefaultsConfigContext(),
+                                            new EnvVarConfigContext(),
+                                            new MapConfigContext(System.getProperties()));
         }
+    }
+
+
+    @CommandLine.Command(name = "connect-test",
+                         header = "Try to connect",
+                         description = "Attempts to connect to Manta using system properties "
+                         + "and environment variables for configuration")
+    public static class ConnectTest extends MantaSubCommand {
+
+        @Override
+        public void run() {
+            final StringBuilder b = new StringBuilder();
+
+            b.append("Creating connection configuration").append(BR);
+            ConfigContext config = buildConfig();
+            b.append(INDENT).append(ConfigContext.toString(config)).append(BR);
 
-        return b.toString();
+            b.append("Creating new connection object").append(BR);
+            try (MantaClient client = new MantaClient(config)) {
+                b.append(INDENT).append(client).append(BR);
+
+                String homeDirPath = config.getMantaHomeDirectory();
+                b.append("Attempting HEAD request to: ").append(homeDirPath).append(BR);
+
+                MantaObjectResponse response = client.head(homeDirPath);
+                b.append(INDENT).append(response).append(BR);
+                b.append("Request was successful");
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+            System.out.println(b.toString());
+        }
     }
 
+
     /**
      * Dumps the configuration used for configuring a Manta client.
      * @return String containing the contents of the parsed configuration
      */
```

Comment:
Did you intend to keep this method JavaDoc? They seem to have been removed from the other commands once they became classes.